tensorflow/contrib/periodic_resample/kernels/periodic_resample_op.h

```diff
@@ -122,36 +230,50 @@ template <class InputDataT,
               input_tensor_shape.dim_size(i), ", desired shape has size ",
               desired_shape[i], "."));
 
-      // target_dimensions[i] = desired_shape(i);
-      target_dimensions[i] = desired_shape[i];
-      new_sliced_size *= target_dimensions[i];
+      (*target_dimensions)[i] = desired_shape[i];
+      new_sliced_size *= (*target_dimensions)[i];
     }
   }
   // at least one index needs to be adjustable
   OP_REQUIRES(context, found,
               tensorflow::errors::InvalidArgument(
                   "periodic_resample expects at least "
                   "one index to be marked as adjustable."));
+  (*target_dimensions)[*adjustable_dimension] =
+      input_tensor_shape.num_elements() / new_sliced_size;
 
-  int count = 0;
-  for (const auto dim_info : input_tensor.shape()) {
-    original_dimensions[count] = dim_info.size;
-    ++count;
-  }
+  *output_size = new_sliced_size * (*target_dimensions)[*adjustable_dimension];
+}
 
-  target_dimensions[adjustable_dimension] = total_size / new_sliced_size;
+// Heuristic number based on measurements on development machine.
```

Comment:
It's fine to measure a heuristic this way (certainly beats guessing), but could you please specify the CPU type of the development machine.

Reply:
Wrote CPU type, also heuristic updated a bit (seems, I checked it on some other computer before  - re-measured it again).